File: app/src/main/java/mega/privacy/android/app/fragments/managerFragments/cu/CuImageViewHolder.java
package mega.privacy.android.app.fragments.managerFragments.cu;

import android.view.View;

import com.bumptech.glide.RequestManager;

import mega.privacy.android.app.databinding.ItemCameraUploadsImageBinding;

/**
 * Created by Piasy{github.com/Piasy} on 2020/7/17.
 */
class CuImageViewHolder extends CuViewHolder {

    private final ItemCameraUploadsImageBinding mBinding;
    private final CuItemSizeConfig mItemSizeConfig;

    public CuImageViewHolder(ItemCameraUploadsImageBinding binding,
            CuItemSizeConfig itemSizeConfig) {
        super(binding.getRoot());
        mBinding = binding;
        mItemSizeConfig = itemSizeConfig;

        setViewSize(binding.getRoot(), binding.icSelected, itemSizeConfig);
    }

    @Override protected void bind(CuNode node, RequestManager requestManager) {
        mBinding.icSelected.setVisibility(node.isSelected() ? View.VISIBLE : View.GONE);

        updateThumbnailDisplay(mBinding.thumbnail, node, mItemSizeConfig, requestManager);
    }

    public ItemCameraUploadsImageBinding binding() {
        return mBinding;
    }
}

File: app/src/main/java/mega/privacy/android/app/fragments/managerFragments/cu/CuImageViewHolder.kt
package mega.privacy.android.app.fragments.managerFragments.cu

import android.view.View
import mega.privacy.android.app.databinding.ItemCameraUploadsImageBinding
import mega.privacy.android.app.gallery.data.GalleryItemSizeConfig

class CuImageViewHolder(
    private val mBinding: ItemCameraUploadsImageBinding,
    private val mItemSizeConfig: GalleryItemSizeConfig
) : CuGridViewHolder(mBinding.root) {

    init {
        setViewSize(mBinding.root, mBinding.icSelected, mItemSizeConfig)
    }

    override fun bind(node: CuNode) {
        mBinding.icSelected.visibility = if (node.isSelected) View.VISIBLE else View.GONE

        updateThumbnailDisplay(
            mBinding.thumbnail,
            node,
            mItemSizeConfig
        )
    }

    fun binding() = mBinding
}